core/src/provider/initiator.rs

```diff
@@ -168,21 +164,23 @@ where
 			genesis: Box::new(Storage::default()),
 			handle,
 			exec,
-			keystore: None,
 			client_config: ClientConfig::default(),
 			pool_config: PoolConfig {
 				is_validator: true,
 				options: Options::default(),
 				revalidation: RevalidationType::Full,
 			},
-			execution_strategies: ExecutionStrategies::default(),
+			execution_extensions: ExecutionExtensions::default(),
 		}
 	}
 
-	/// Overwrites the used `ExecutionStrategies` that will be used when initiating the
+	/// Overwrites the used `ExecutionExtensions` that will be used when initiating the
 	/// structs for a core builder.
-	pub fn with_exec_strategies(&mut self, execution_strategies: ExecutionStrategies) -> &mut Self {
-		self.execution_strategies = execution_strategies;
```

Comment:
I would also not remove that but rather have both of the methods availabvle here.

Exeutions strategies is also part of `ExecutionExtension`